File: BOJ/[1328] 고층 빌딩/Johoseong/1328.py
import sys
input = sys.stdin.readline

N, L, R = map(int, input().split())
dp = [[[0 for _ in range(N + 1)] for _ in range(N + 1)] for _ in range(N + 1)]
dp[1][1][1] = 1
for i in range(2, N + 1):
    for j in range(1, L + 1):
        for k in range(1, R + 1):
            dp[i][j][k] = dp[i - 1][j][k] * (i - 2) + dp[i - 1][j - 1][k] + dp[i - 1][j][k - 1]
            dp[i][j][k] %= 1000000007

print(dp[N][L][R])

File: BOJ/[1328] 고층 빌딩/woong-jae/1328.js
const readline = require('readline');
const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

const input = [];
rl.on('line', function (line) {
    input.push(line.split(' '));
}).on("close", function () {
    solution(...input[0].map(value => parseInt(value)));
    process.exit();
});

function solution(N, L, R) {
    const mod = 1000000007;

    const dp = Array.from(Array(N + 1), () => Array.from(Array(N + 1), () => Array(N + 1).fill(0)));
    dp[1][1][1] = 1;
    for(let length = 2; length <= N; length++) {
        for(let l = 1; l <= length; l++) {
            for(let r = 1; r <= length; r++) {
                dp[length][l][r] = (dp[length - 1][l - 1][r] + dp[length - 1][l][r - 1] + dp[length - 1][l][r] * (length - 2)) % mod;
            }
        }
    }

    console.log(dp[N][L][R]);
}